crates/ruff_python_semantic/src/cfg/graph.rs

```diff
@@ -0,0 +1,282 @@
+use ruff_index::{newtype_index, IndexVec};
+use ruff_python_ast::Stmt;
+use ruff_text_size::{Ranged, TextRange};
+use smallvec::{smallvec, SmallVec};
+
+/// Returns the control flow graph associated to an array of statements
+pub fn build_cfg(stmts: &[Stmt]) -> CFG<'_> {
+    let mut builder = CFGBuilder::with_capacity(stmts.len());
+    builder.process_stmts(stmts);
+    builder.finish()
+}
+
+/// Control flow graph
```

Comment:
It would be great to have some module or struct level documentation explaining what a CFG is, what are basic blocks, how are they connected, and how the data is represented. 

You may want to wait with writing this documentation if we end up changing the representation of `Nodes` (`Block`) and `Edges`